src/plugins/credential_management/server/plugin.ts

```diff
@@ -20,34 +21,43 @@ import {
 import { CredentialManagementPluginSetup, CredentialManagementPluginStart } from './types';
 import { registerRoutes } from './routes';
 import { credentialSavedObjectType } from './saved_objects';
+import { ConfigSchema } from '../config';
+import { CryptoCli } from './crypto';
 
 export class CredentialManagementPlugin
   implements Plugin<CredentialManagementPluginSetup, CredentialManagementPluginStart> {
   private readonly logger: Logger;
+  private initializerContext: PluginInitializerContext<ConfigSchema>;
 
-  constructor(initializerContext: PluginInitializerContext) {
+  private cryptoCli?: CryptoCli;
+
+  constructor(initializerContext: PluginInitializerContext<ConfigSchema>) {
     this.logger = initializerContext.logger.get();
+    this.initializerContext = initializerContext;
   }
 
-  public setup(core: CoreSetup) {
-    this.logger.debug('credentialManagement: Setup');
-    const router = core.http.createRouter();
-
-    // Register server side APIs
-    registerRoutes(router);
-
-    // Register credential saved object type
-    core.savedObjects.registerType(credentialSavedObjectType);
-
+  public async setup(core: CoreSetup) {
+    this.logger.debug('credential_management: Setup');
+    const { enabled, materialPath } = await this.initializerContext.config
```

Comment:
actually, if you set plugin config `enabled` to false, the plugin will not be loaded by plugin service, thus it won't initialize the plugin